components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectSystemClaimImpl.java

```diff
@@ -145,8 +149,23 @@ private boolean isCodeHashApplicable(String responseType) {
 
     private boolean isAccessTokenHashApplicable(String responseType) {
         // At_hash is generated on an access token. Therefore check whether the response type returns an access_token.
-        // id_token and none response types don't return and access token
-        return !OAuthConstants.ID_TOKEN.equalsIgnoreCase(responseType) &&
-                !OAuthConstants.NONE.equalsIgnoreCase(responseType);
+        for (String respType : getResponseTypes(responseType)) {
+            if (OAuthConstants.TOKEN.equalsIgnoreCase(respType)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private String[] getResponseTypes(String responseType) {
+
+        if (isNotBlank(responseType)) {
+            return responseType.split(" ");
```

Comment:
Just for the sake of argument can there be a case, where the response_type be separated by more than one space? or will it be considered as invalid at a layer above here?

Reply:
something like
`response_type=code  id_token`

Reply:
It will be handled before coming to this point and will give the following response:
`https://localhost.com:9443/oauth2/a?error_description=invalid_request%2C+Invalid+response_type+parameter+value&state=Sr7tArZM9x&error=invalid_request`